src/Core/Managed/Shared/Extensibility/Implementation/OperationHolder.cs

```diff
@@ -74,17 +78,38 @@ protected virtual void Dispose(bool disposing)
                     if (!this.isDisposed)
                     {
                         var operationTelemetry = this.Telemetry;
+                        bool isActivityEnabled = false;
+#if !NET40
+                        isActivityEnabled = ActivityProxy.TryRun(() =>
```

Comment:
Is this doing a runtime availability check for the DiagnosticsSource assembly?
If so, consider making ```isActivityEnabled``` a static field so it's initialized just once and not each time this is called.